lib/event_grid.jl

```diff
@@ -0,0 +1,75 @@
+
+###########################################
+## Interpolated event grid (requires Grid)
+###########################################
+
+@comment """
+# Interpolated event grid
+"""
+
+using Requires
+
+@require Grid begin
+
+    using Grid
+    
+    function search_grid (g::InterpGrid, x::Float64)
+        i = searchsortedfirst (g, x)
+        g[i]
+    end
+
+    function grid_point (x::Float64, v::Float64)
+        x - v
+    end
+    
+@doc* """
+Interpolated event grid (requires Grid library).
+
+```julia
+
+EventGrid(events, tf, dt)
+```
+
+Given a vector with event times and a sampling step, creates a
+continuous interpolated function that has a negative value between
+events, and crosses zero at each event time.
+The function `grid_input` can be used to detect events in model equations.
+
+### Arguments
+
+* `event` : Vector of event times
+* `tf` : grid end time
+* `dt` : event grid sampling time
+
+### Example
+
+```julia
+
+input = EventGrid([1.0,2.0,3.0],tstop,0.1)
+
+Event(grid_input(input),
+      Equation[
+                reinit(x, x + k)
+              ],
+      Equation[])
+
+```
+""" ->
+
+function EventGrid(events,tf,dt)
+    
+    g0 = InterpGrid(events, 0.0, InterpNearest)
```

Comment:
For methods from Grid, it'd be good to fully qualify them. It's not absolutely necessary, but it might help reduce conflicts in the future, and it's clearer where they come from.